server/channels/api4/post.go

```diff
@@ -652,7 +665,17 @@ func deletePost(c *Context, w http.ResponseWriter, _ *http.Request) {
 		}
 	}
 
-	if _, err := c.App.DeletePost(c.AppContext, c.Params.PostId, c.AppContext.Session().UserId); err != nil {
+	if permanent {
+		if *c.App.Config().ServiceSettings.EnableAPIPostDeletion {
```

Comment:
3/5 that this check should happen before we load the post from the DB.

Reply:
Moved this up above the GetPost call along with the session check